script/zonemaster-cli

```diff
@@ -7,16 +7,51 @@ use Zonemaster::CLI;
 use File::Spec;
 use autodie;
 
+# Load default arguments from file in home directory, if any
+# This must be loaded before any global file to make the local
+# file take precedence
 my $homedir  = ((getpwuid($<))[7]) || $ENV{HOME};
-my $conffile = File::Spec->catfile($homedir, '.zonemaster', 'cli.args');
+my $homeconffile = File::Spec->catfile($homedir, '.zonemaster', 'cli.args');
+
+if (-r $homeconffile) {
+    open my $fh, '<', $homeconffile;
+    my @lines;
+    while (<$fh>) {
+        chomp;
+        next if /^\s*$/;
+        next if /^\s*#/;
+        push @lines, $_;
+    };
+    unshift @ARGV, @lines;
+}
+
+# Load default arguments from global file, if any
+my @globalconf = (
+    '/etc/zonemaster/cli.args',
+    '/usr/local/etc/zonemaster/cli.args'
+    ); # Order is significant.
+my $globalconffile;
+
+for my $p (@globalconf) {
+    if ( -e $p and -r $p ) {
+        $globalconffile = $p;
+        last;
+    }
+}
 
-if (-r $conffile) {
-    open my $fh, '<', $conffile;
-    my @lines = $fh->getlines;
-    chomp(@lines);
+if ( defined $globalconffile ) {
+    open my $fh, '<', $globalconffile;
+    my @lines;
+    while (<$fh>) {
+        chomp;
+        next if /^\s*$/;
+        next if /^\s*#/;
+        push @lines, $_;
+    };
```

Comment:
Did you consider making a function to avoid code duplication when reading each file?

Reply:
Fixed.